packages/router/src/grpc/custody/authorize.ts

```diff
@@ -15,6 +18,19 @@ export const authorize: Impl['authorize'] = async (req, ctx) => {
   const sess = ctx.values.get(extSessionCtx);
   const local = ctx.values.get(extLocalCtx);
 
+  const services = ctx.values.get(servicesCtx);
+  const walletServices = await services.getWalletServices();
+  const assetsMetadata = await walletServices.indexedDb.getAllAssetsMetadata();
+  const denomMetadataByAssetId = assetsMetadata.reduce<Record<string, DenomMetadataAsJson>>(
```

Comment:
request the metadata after the throws that are close to the approveReq call, we might bail out before we need any of it

Reply:
done